utils/utils.go

```diff
@@ -43,3 +44,17 @@ func GetQueryParamListValues(queryVars url.Values, varKey string, maxNumItems in
 	}
 	return items, nil
 }
+
+// utility function to cut a slice according to the provided offset and limit.
+// limit=0 means no limit, and values higher than the slice length are ignored
```

Comment:
limit set to 0 should mean 0 items, if a user does not want a limit and that is a feature that we need to be handling then I would suggest using -1 at which point this needs to be documented in the swagger spec and handling of negative values has an exception on -1